src/mscorlib/shared/System/Collections/Generic/Dictionary.cs

```diff
@@ -143,7 +143,7 @@ public IEqualityComparer<TKey> Comparer
         {
             get
             {
-                return _comparer;
+                return (_comparer is NonRandomizedStringEqualityComparer) ? (IEqualityComparer<TKey>)EqualityComparer<string>.Default : _comparer;
```

Comment:
I'm surprised this compiles.  Shouldn't it be `EqualityComparer<TKey>` rather than `EqualityComparer<string>`, or am I missing something?

Reply:
Ah, I see why it works ok... TKey must be a string for _comparer to get set to NonRandomizedStringEqualityComparer.

Reply:
actually compile also with TKey because the constraints are respected and for now we've only optimization with string TKey, i think with EqualityComparer<string> the intentions are more clear. Do you agree?